### DISCUSSION TITLE
@aware tag properties break on rerender

### DISCUSSION THREAD

**Discussion by Persaeus** [livewire/livewire#4859] [7. Bugs (archived)] [answered]
The `@aware` tag appears to not be working correctly when inside a Livewire component.

On initial render, it appears to be working fine: the `@aware` tag fills the component with properties of parent components. However, when a component is re-rendered, it is rendered as if parent properties were not given.

Looking deeper, it appears that the `$componentStack` and `$componentData` properties on the view factory (`Illuminate\View\Factory`) aren't available when a view is re-rendered standalone. These are needed for the `@aware` tag to inherit parent properties.

```blade
<!-- /resources/views/app.blade.php -->

<x-wrapper color="bg-blue-500">
    <livewire:foobar />
    ...
</x-wrapper>
```

```blade
<!-- /resources/views/livewire/foobar.blade.php -->

<x-border />
```

```blade
<!-- /resources/views/components/border.blade.php -->

@aware(['color' => 'bg-red-500'])
<div class="{{ $color }} w-10 h-10"></div>
```

In the above example, on initial render, the border child-element correctly inherits from the parent wrapper element and is blue. On re-render, it changes to the default red

**Comment by joshhanley** [comment, marked as answer]
@nihilsen `@aware` wasn't designed for use with Livewire components, only blade components. So personally wouldn't be using it with a Livewire component.

Hope this helps!

**Comment by satoved** [comment]
@nihilsen on update, Livewire component FooBar would be rendered separetely from a livewire.message route, how should it know about the outer variables again? You'll have to pass them yourself :(

**Reply by joshhanley** [reply]
@satoved yep, that's correct 🙂 that's what I meant above by it wasn't designed to work with Livewire. On subsequent requests, it wouldn't know what the values of the outer variables are as they don't exist in the `livewire.message` route context.

**Comment by Persaeus** [comment]
> @nihilsen `@aware` wasn't designed for use with Livewire components, only blade components.

That was my observation as well. This is an isolated case anyways, as the `@aware` appears to work fine when completely enclosed in a Livewire component.
